### PR TITLE
Add OTLP HTTP Log Exporter example using tracing appender.

### PR DESCRIPTION
**Pull request by lalitb**
## Changes
- Update the existing OTLP HTTP exporter to export logs instrumented through tokio library, using tracing appender. This change could have suffered from infinite live-loop issue described in #1171, but the Hyper HTTP library emits tracing/debug logs only if the tracing feature (which is disabled by default) is enabled for this library, and Cargo.toml for this example doesn't enable it. Also, there are no errors reported from this library (which would also get routed to OpenTelemetry appender) if the steps in examples are executed properly. So only the logs, traces and metrics geerated from the example code are exported to the collector.

- Also for simplicity removed all-in-one jaeger exporter from the collector configuration, and instead used Debug Exporter which dumps all events to console.

- The infinite live-issue discussed in #1171 need to be investigated separately, one option could be to filter all such non-required events using `event_enabled` function by checking the target/module.

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-otlp/examples/basic-otlp-http/src/main.rs
Comment (shaun-cox) [inline opentelemetry-otlp/examples/basic-otlp-http/src/main.rs:24]: Consider passing `env!("CARGO_BIN_NAME")` instead of hard-coding the example name, as that demonstrates best practice and works if someone copies the example and renames their binary.

**Review by shaun-cox — CHANGES_REQUESTED** [review]
Thanks for doing this.  Just one nit which would be nice to address before merging...

File: opentelemetry-otlp/examples/basic-otlp-http/src/main.rs
Comment (lalitb) [reply opentelemetry-otlp/examples/basic-otlp-http/src/main.rs]: Good point, using hardcoded name now only if the env-variable is not set:
```rust
    let service_name = std::env::var("CARGO_BIN_NAME")
        .unwrap_or_else(|_| String::from("basic-otlp-http-logging-example"));
```

**Comment by lalitb** [comment]
The CI is failing with:

``
error: package `regex-syntax v0.8.1` cannot be built because it requires rustc 1.65 or newer, while the currently active rustc version is 1.64.0
Error: Process completed with exit code 101.
``

Does it mean we need to bump up msrv to 1.65.0 ?

File: opentelemetry-otlp/examples/basic-otlp-http/README.md
Comment (cijothomas) [inline opentelemetry-otlp/examples/basic-otlp-http/README.md:14]:
```suggestion
The docker-compose terminal will display logs, metrics and traces.
```

File: opentelemetry-otlp/examples/basic-otlp-http/docker-compose.yaml
Comment (cijothomas) [inline opentelemetry-otlp/examples/basic-otlp-http/docker-compose.yaml:13]: we don't need gRPC as this is only for http.. maybe better to combine both to a single example, as switching from http to gRPC is a small change in exporter.

Comment (lalitb) [reply opentelemetry-otlp/examples/basic-otlp-http/docker-compose.yaml:13]: Yes, this PR is meant to add the log exporter in existing OTLP HTTP example. We can consolidate both the examples into single as separate cleanup PR.

Comment (lalitb) [reply opentelemetry-otlp/examples/basic-otlp-http/docker-compose.yaml]: Have removed the unwanted ports from both the examples now.

**Review by cijothomas — APPROVED** [review]

**Comment by lalitb** [comment]
> The CI is failing with:
> 
> `` error: package `regex-syntax v0.8.1` cannot be built because it requires rustc 1.65 or newer, while the currently active rustc version is 1.64.0 Error: Process completed with exit code 101. ``
> 
> Does it mean we should bump up msrv to 1.65.0 ?

@jtescher @TommyCpp  Seems the PRs are blocked on this error. Should we raise a PR to bump the msrv version ?

**Comment by jtescher** [comment]
@lalitb We could do that, an alternative would be to swap the view regex impl to use a glob impl instead as suggested in https://github.com/open-telemetry/opentelemetry-rust/issues/1262, which would drop the need for the dependency entirely

File: opentelemetry-otlp/examples/basic-otlp/src/main.rs
Comment (shaun-cox) [inline opentelemetry-otlp/examples/basic-otlp/src/main.rs:55]: sorry for the confusion earlier, but I meant use the standard [`env!`](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates) to get this.  e.g. just
``` rust
let service_name = env!("CARGO_BIN_NAME");
```

**Review by shaun-cox — CHANGES_REQUESTED** [review]

File: opentelemetry-otlp/examples/basic-otlp/src/main.rs
Comment (lalitb) [reply opentelemetry-otlp/examples/basic-otlp/src/main.rs]: Yes my bad, this variable will not be available at run-time. Fixed it accordingly.

**Review by shaun-cox — APPROVED** [review]

**Pull request merged**
